File: core/src/com/alta_v2/physics/executionContext/reserveData/ReservableActor.java
package com.alta_v2.physics.executionContext.reserveData;

import com.alta_v2.physics.executionContext.Reservable;
import lombok.Getter;

/**
 * Describes value that is distributable for read/write access.
 */
public class ReservableActor extends Reservable<Boolean> {

    @Getter
    private final int id;

    @Getter
    private final ReservablePoint globalPoint;

    @Getter
    private final ReservablePoint localPoint;

    @Getter
    private final ReservablePersonView view;

    @Getter
    private final ReservableBoolean isMoving;

    /**
     * Initialize new instance of {@link ReservableActor}.
     * @param id - the identifier of actor.
     */
    public ReservableActor(int id) {
        this.id = id;
        this.globalPoint = new ReservablePoint();
        this.view = new ReservablePersonView();
        this.isMoving = new ReservableBoolean();
        this.localPoint = new ReservablePoint();
        this.value = false;
    }
}

File: core/src/com/alta_v2/physics/executionContext/reserveData/ReservableActor.kt
package com.alta_v2.physics.executionContext.reserveData

import com.alta_v2.physics.executionContext.Reservable

/**
 * Describes value that is distributable for read/write access.
 */
class ReservableActor(val id: Int) : Reservable<Boolean>() {

    val globalPoint = ReservablePoint()
    val localPoint = ReservablePoint()
    val view = ReservablePersonView()
    val isMoving: ReservableBoolean = ReservableBoolean()

    init {
        value = false
    }

    fun getValue() = value
}